dumnati/src/scraper.rs

```diff
@@ -5,6 +5,9 @@ use reqwest::Method;
 use std::num::NonZeroU64;
 use std::time::Duration;
 
+/// Default timeout for HTTP requests (4 hours).
+const DEFAULT_HTTP_REQ_TIMEOUT: Duration = Duration::from_secs(4 * 60 * 60);
```

Comment:
Can you elaborate on the logic for selecting 4 hours here? Surely if the server didn't reply after even e.g. 15m, it's unlikely to ever reply, right?

If we're going to exit and get restarted by the orchestrator anyway, we might as well fail earlier, no? (A related discussion though is reporting this somehow to some monitoring service.)

Reply:
It's just an arbitrary overlarge value which in my brain I can safely approximate to "insanely long".

But notably, this is a **completion** timeout not a first-reply one. That is, it is an hard arbitrary upper limit we are placing on how big/slow a positive response can be.

Plus, we are not going to exit but just proceed with the next scraping loop, increasing the error-counter metrics (we don't track this metrics right now).

Reply:
From an out-of-band discussion: we expect the responses to be short and quick to complete, as we are only grabbing JSON metadata. 4h is overly large, to the point that it impacts our release work if the backend is hanging for such a long time. Going for 30 minutes instead (@56kbps it would still fit 12 MiB).